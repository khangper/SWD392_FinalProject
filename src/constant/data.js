export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
