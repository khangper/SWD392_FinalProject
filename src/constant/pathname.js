export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
