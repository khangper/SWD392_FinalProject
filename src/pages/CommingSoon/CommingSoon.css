.cs-time {
    display: flex;
    align-items: center;
    justify-content: center;
    
}

.cs_coming_title {
    font-size: 80px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    line-height: 1em;
    text-shadow: 1px 10px 6px rgba(0, 0, 0, 0.2); /* Ensure this is the correct property */
    padding-bottom: 29px;
}

.cs-timer {
    display: flex;
    flex-direction: column;
    vertical-align: middle;
    text-align: center;
    text-transform: uppercase;
    width: 16%;
}

.cs-thnk_title1 {
    font-size: 24px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-shadow: 1px 10px 6px rgba(0, 0, 0, 0.2);
    margin-top: 20px;
}

.cs-search {
    display: flex;
    justify-content: center;
    align-items: center;
    
    padding: 20px;
}

.cs-search-box {
    display: flex;
}

.cs-search-input {
    display: flex;
    border-radius: 4px;
    overflow: hidden;
}

.cs-input {
    padding-left: 2em !important;
    padding-right: 8em !important;
    height: 50px !important;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    border: 2px solid #fff !important;
}

.cs-submit {
    background-color: #e53935; /* Màu nút */
    color: white;
    border: none;
    padding: 10px 20px;
    cursor: pointer;
    transition: background-color 0.3s;
}

.cs-submit:hover {
    background-color: #d32f2f; /* Màu khi hover */
}
