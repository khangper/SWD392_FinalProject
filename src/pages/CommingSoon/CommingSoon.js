import React, { useEffect, useState } from 'react'
import './CommingSoon.css'
import { Link } from 'react-router-dom'
const CommingSoon = () => {
    const calculateTimeLeft = () => {
        const difference = +new Date('2025-01-01') - +new Date();
        let timeLeft = {};

        if (difference > 0) {
            timeLeft = {
                days: Math.floor(difference / (1000 * 60 * 60 * 24)),
                hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
                minutes: Math.floor((difference / 1000 / 60) % 60),
                seconds: Math.floor((difference / 1000) % 60)
            };
        }

        return timeLeft;
    };

    const [timeLeft, setTimeLeft] = useState(calculateTimeLeft());

    useEffect(() => {
        const timer = setTimeout(() => {
            setTimeLeft(calculateTimeLeft());
        }, 1000);

        return () => clearTimeout(timer);
    }, [timeLeft]);



    return (
        <div className="Thanks-wrapper">
            {/* <div className="container"> */}
            <div className="Thanks-row">
                <div className="Thanks-col-md-12">
                    <div className="Thanks-cmtk_group">
                        <Link to="/home">
                            <div className="Thanks-ct-logo">
                                <a href="index.html"><img src="https://gambolthemes.net/html-items/cursus-new-demo/images/ct_logo.svg" alt="" /></a>
                            </div>
                        </Link>

                        <div className="Thanks-cmtk_dt">
                            {/* <div className='cs-time'>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        335 :
                                    </span>
                                    <label>Days</label>
                                </div>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        00 :
                                    </span>
                                    <label>Hours</label>
                                </div>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        00 :
                                    </span>
                                    <label>Minutes</label>
                                </div>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        00
                                    </span>
                                    <label>Seconds</label>
                                </div>
                            </div> */}
                            <div className='cs-time'>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        {timeLeft.days || '0'} :
                                    </span>
                                    <label>Days</label>
                                </div>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        {timeLeft.hours || '00'} :
                                    </span>
                                    <label>Hours</label>
                                </div>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        {timeLeft.minutes || '00'} :
                                    </span>
                                    <label>Minutes</label>
                                </div>
                                <div className='cs-timer'>
                                    <span className="cs_coming_title">
                                        {timeLeft.seconds || '00'}
                                    </span>
                                    <label>Seconds</label>
                                </div>
                            </div>
                            <h4 className="cs-thnk_title1">We will be comming soon!</h4>
                            <div class="cs-search">
                                <form>
                                    <div class="cs-search-box">
                                        <div class="cs-search-input">
                                            <input class="cs-input" type="email" name="emailaddress" value="" id="id_email" required="" maxlength="64" placeholder="Email address" />
                                            <button class="cs-submit" type="submit">Notify Me</button>
                                        </div>
                                    </div>
                                </form>
                            </div>
                        </div>
                        <div className="Thanks-tc_footer_main">
                            <div className="Thanks-tc_footer_left">
                                <ul>
                                    <li><a href="about_us.html">About</a></li>
                                    <li><a href="press.html">Press</a></li>
                                    <li><a href="contact_us.html">Contact Us</a></li>
                                    <li><a href="coming_soon.html">Advertise</a></li>
                                    <li><a href="coming_soon.html">Developers</a></li>
                                    <li><a href="terms_of_use.html">Copyright</a></li>
                                    <li><a href="terms_of_use.html">Privacy Policy</a></li>
                                    <li><a href="terms_of_use.html">Terms</a></li>
                                </ul>
                            </div>
                            <div className="Thanks-tc_footer_right">
                                <p>Â© 2024 <strong>Cursus</strong>. All Rights Reserved.</p>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

        </div>
    )
}

export default CommingSoon