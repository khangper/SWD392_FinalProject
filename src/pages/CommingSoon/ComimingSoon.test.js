import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import CommingSoon from "./CommingSoon";
import { act } from "react-dom/test-utils";
import { screen, waitFor } from "@testing-library/react";

describe("CommingSoon", () => {
    let dateNowSpy;

    beforeAll(() => {
        // Mock Date.now to return a consistent time for snapshots
        const fixedDate = new Date('2024-07-01T00:00:00Z');
        dateNowSpy = jest.spyOn(global.Date, 'now').mockImplementation(() => fixedDate.getTime());

        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        dateNowSpy.mockRestore();
        console.error.mockRestore();
        console.log.mockRestore();
    });

    it("renders the CommingSoon component", async () => {
        let container;

        await act(async () => {
            const result = renderWithProviders(<CommingSoon />);
            container = result.container;
        });

        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
});
