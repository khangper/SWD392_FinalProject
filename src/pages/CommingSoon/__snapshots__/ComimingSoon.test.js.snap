// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommingSoon renders the CommingSoon component 1`] = `
<div>
  <div
    class="Thanks-wrapper"
  >
    <div
      class="Thanks-row"
    >
      <div
        class="Thanks-col-md-12"
      >
        <div
          class="Thanks-cmtk_group"
        >
          <a
            href="/home"
          >
            <div
              class="Thanks-ct-logo"
            >
              <img
                alt=""
                src="https://gambolthemes.net/html-items/cursus-new-demo/images/ct_logo.svg"
              />
            </div>
          </a>
          <div
            class="Thanks-cmtk_dt"
          >
            <div
              class="cs-time"
            >
              <div
                class="cs-timer"
              >
                <span
                  class="cs_coming_title"
                >
                  155
                   :
                </span>
                <label>
                  Days
                </label>
              </div>
              <div
                class="cs-timer"
              >
                <span
                  class="cs_coming_title"
                >
                  13
                   :
                </span>
                <label>
                  Hours
                </label>
              </div>
              <div
                class="cs-timer"
              >
                <span
                  class="cs_coming_title"
                >
                  54
                   :
                </span>
                <label>
                  Minutes
                </label>
              </div>
              <div
                class="cs-timer"
              >
                <span
                  class="cs_coming_title"
                >
                  10
                </span>
                <label>
                  Seconds
                </label>
              </div>
            </div>
            <h4
              class="cs-thnk_title1"
            >
              We will be comming soon!
            </h4>
            <div
              class="cs-search"
            >
              <form>
                <div
                  class="cs-search-box"
                >
                  <div
                    class="cs-search-input"
                  >
                    <input
                      class="cs-input"
                      id="id_email"
                      maxlength="64"
                      name="emailaddress"
                      placeholder="Email address"
                      type="email"
                    />
                    <button
                      class="cs-submit"
                      type="submit"
                    >
                      Notify Me
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
          <div
            class="Thanks-tc_footer_main"
          >
            <div
              class="Thanks-tc_footer_left"
            >
              <ul>
                <li>
                  <a
                    href="about_us.html"
                  >
                    About
                  </a>
                </li>
                <li>
                  <a
                    href="press.html"
                  >
                    Press
                  </a>
                </li>
                <li>
                  <a
                    href="contact_us.html"
                  >
                    Contact Us
                  </a>
                </li>
                <li>
                  <a
                    href="coming_soon.html"
                  >
                    Advertise
                  </a>
                </li>
                <li>
                  <a
                    href="coming_soon.html"
                  >
                    Developers
                  </a>
                </li>
                <li>
                  <a
                    href="terms_of_use.html"
                  >
                    Copyright
                  </a>
                </li>
                <li>
                  <a
                    href="terms_of_use.html"
                  >
                    Privacy Policy
                  </a>
                </li>
                <li>
                  <a
                    href="terms_of_use.html"
                  >
                    Terms
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="Thanks-tc_footer_right"
            >
              <p>
                Â© 2024 
                <strong>
                  Cursus
                </strong>
                . All Rights Reserved.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
